lib/ansible/modules/cloud/amazon/iam_user_facts.py

```diff
@@ -0,0 +1,181 @@
+extends_documentation_fragment:
+  - aws
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+  # Gather facts about "test" user.
+  - name: Get IAM user facts
+    iam_user_facts:
+      name: "test"
+
+  # Gather facts about all users in the "dev" group.
+  - name: Get IAM user facts
+    iam_user_facts:
+      group: "dev"
+
+  # Gather facts about all users with "dev" path.
+  - name: Get IAM user facts
+    iam_user_facts:
+      path: "dev"
+'''
+
+RETURN = '''
+arn:
+    description: the ARN of the user
+    returned: if user exists
+    type: string (ARN)
+    sample: "arn:aws:iam::156360693172:user/dev/test_user"
+create_date:
+    description: the datetime user was created
+    returned: if user exists
+    type: date (UTC)
+    sample: "2016-05-24T12:24:59+00:00"
+password_last_used:
+    description: the last datetime the password was used by user
+    returned: if password was used at least once
+    type: date (UTC)
+    sample: "2016-05-25T13:39:11+00:00"
+path:
+    description: the path to user
+    returned: if user exists
+    type: string
+    sample: "/dev/"
+user_id:
+    description: the unique user id
+    returned: if user exists
+    type: string
+    sample: "AIDUIOOCQKTUGI6QJLGH2"
+user_name:
+    description: the user name
+    returned: if user exists
+    type: string
+    sample: "test_user"
+'''
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, ParamValidationError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict()
+    iam_users = []
+
+    if name and not path:
+        params['UserName'] = name
+        try:
+            iam_users.append(connection.get_user(**params)['User'])
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    if group:
+        params['MaxItems'] = 1000
+        params['GroupName'] = group
+        try:
+            iam_users = connection.get_group(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    if path and not group:
+        params['MaxItems'] = 1000
+        params['PathPrefix'] = path
+        try:
+            iam_users = connection.list_users(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
```

Comment:
Can you also catch generic exception here and raise the traceback like `module.fail_json(msg="Failed while listing IAM users", exception=traceback.format_exc())`?